src/v/iceberg/conversion/values_protobuf.cc

```diff
@@ -320,6 +403,11 @@ ss::future<optional_value_outcome> single_field_to_value(
           == pb::Descriptor::WELLKNOWNTYPE_TIMESTAMP) {
             co_return co_await convert_timestamp(std::move(msg_field));
         }
+        if (
+          field_descriptor.message_type()->well_known_type()
+          == pb::Descriptor::WELLKNOWNTYPE_STRUCT) {
+            co_return co_await convert_struct_to_json(std::move(msg_field));
```

Comment:
since there is recursion here, we need to pass `stack` and enforce depth checks so we don't stack overflow